### PR TITLE
`CI`: also run tests on `watchOS`

### PR DESCRIPTION
**Pull request by NachoSoto**
This is a follow up to #717.
In order to be able to reproduce #2338, which seems to be an issue only on `watchOS`, it's important that we now run tests on `watchOS` as well.
We probably won't be able to fully replicate that issue since we don't run tests on actual devices, so they'll still run on 64 bits. But this way we might be able to prevent issues that are only reproducible on `watchOS` in the future.

## Changes:
- Added CircleCI job
- Added lane
- Simplified `UnitTestsHostApp` definition using `SwiftUI`
- Increased _test only_ `watchOS` deployment target to 7.0 (this was necessary to be able to have a common `@main` for the host app)
- Increased _test only_ `tvOS` deployment target to `14.0` (same as above, plus we only run tvOS tests on the latest version anyway. We test backwards compatibility through iOS already)
- Fixed test compilation on `watchOS`
- Disabled `HTTPClient` tests on `watchOS` (see https://github.com/AliSoftware/OHHTTPStubs/issues/287)
- `StoreKitConfigTestCase` is only available from `watchOS 7.0` since that's when `SKTestSession` was introduced
- Added `watchOS` support to `UnitTests` target
- Added `watchOS` support to `ReceiptParserTests` target
- Fixed `isFamilySharable` `watchOS` version checks for version 8.0
- Created new `CI-RevenueCat` Test Plan to only run unit tests and not StoreKit tests on `watchOS`

### CODE REVIEW

File: Tests/StoreKitUnitTests/LocalReceiptParserStoreKitTests.swift [hunk 1]
```diff
@@ -17,7 +17,7 @@ import Nimble
 import StoreKit
 import XCTest
 
-@available(iOS 14.0, tvOS 14.0, macOS 11.0, watchOS 6.2, *)
+@available(iOS 14.0, tvOS 14.0, macOS 11.0, watchOS 7.0, *)
```
Comment (aboedo) [inline Tests/StoreKitUnitTests/LocalReceiptParserStoreKitTests.swift:20]: oh, [`SKTestSession` is watchOS 7+](https://developer.apple.com/documentation/storekittest/sktestsession#), I'm surprised this even worked before

File: Tests/StoreKitUnitTests/StoreProductTests.swift [hunk 2]
```diff
@@ -107,14 +107,17 @@ class StoreProductTests: StoreKitConfigTestCase {
...
         expect(storeProduct.localizedTitle) == "Monthly Free Trial"
         // open the StoreKit Config file as source code to see the expected value
         expect(storeProduct.subscriptionGroupIdentifier) == "7096FF06"
 
         expect(storeProduct.subscriptionPeriod?.unit) == .month
         expect(storeProduct.subscriptionPeriod?.value) == 1
 
+        if #available(iOS 14.0, macOS 11.0, tvOS 14.0, watchOS 8.0, *) {
+            expect(storeProduct.isFamilyShareable) == true
+        }
```
Comment (aboedo) [inline Tests/StoreKitUnitTests/StoreProductTests.swift:119]: we don't need to fix this one right now, but we may have messed up here - [`isFamilyShareable` is available on watchOS 7](https://developer.apple.com/documentation/storekit/skproduct/3564805-isfamilyshareable#), not sure why we limited to 8 in StoreProduct but it was probably an oversight

File: Tests/UnitTests/Networking/HTTPClientTests.swift [hunk 3]
```diff
@@ -27,6 +27,11 @@ class BaseHTTPClientTests: TestCase {
     override func setUpWithError() throws {
         try super.setUpWithError()
 
+        #if os(watchOS)
+        // See https://github.com/AliSoftware/OHHTTPStubs/issues/287
+        try XCTSkipIf(true, "OHHTTPStubs does not currently support watchOS")
+        #endif
```
Comment (aboedo) [inline Tests/UnitTests/Networking/HTTPClientTests.swift:33]: would love to replace OHHTTPStubs altogether some day

**Review by aboedo — APPROVED** [review]

File: Tests/UnitTests/Networking/HTTPClientTests.swift (hunk 3 above)
Comment (NachoSoto) [reply Tests/UnitTests/Networking/HTTPClientTests.swift:33]: Yup..

File: Tests/StoreKitUnitTests/StoreProductTests.swift (hunk 2 above)
Comment (NachoSoto) [reply Tests/StoreKitUnitTests/StoreProductTests.swift:119]: Huh thanks! I thought I saw this was watchOS 8.0 on `StoreKit` itself. I'll fix it!

File: Tests/StoreKitUnitTests/LocalReceiptParserStoreKitTests.swift (hunk 1 above)
Comment (NachoSoto) [reply Tests/StoreKitUnitTests/LocalReceiptParserStoreKitTests.swift:20]: We never compiled this on `watchOS`.

**Pull request merged**